### ISSUE TITLE
no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beata1"

### ISSUE THREAD

**Issue by persistman** [kubeflow/manifests#2028]
istio: 1.9.6
k8s: 1.22.1
kustomize: 4.3.0
 i installed the istio crds with command 
`kustomize build common/istio-1-9/istio-crds/base | kubectl apply -f -`
but it iis error like this
`
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beata1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
unable to recognize "STDIN": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
`
I try to change  v1beata1 to v1, but still error,and the error llike this:
`error: error validating "STDIN": error validating data: [ValidationError(CustomResourceDefinition.spec): unknown field "subresources" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionSpec, ValidationError(CustomResourceDefinition.spec): unknown field "validation" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionSpec]; if you choose to ignore these errors, turn validation off with --validate=false`

so i had to change the version back.
i do not know why, pls help!

**Comment by rc-coderepo** [comment]
**apiextensions.k8s.io/v1beata1** is deprecated in k8s v1.22.0 version. 
[](https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/#:~:text=The%20beta%20CustomResourceDefinition%20API%20(apiextensions.k8s.io/v1beta1))

Try with k8s - 1.20.x  and kustomize: 3.2.0. Follow v1.3.1 tag documentation [](https://github.com/kubeflow/manifests/tree/v1.3.1) it should work.

**Comment by HWiese1980** [comment]
What do if you're trying to run this on Docker Desktop where you don't have much of a choice which Kubernetes version is running?

**Comment by taghoutigomycode** [comment]
> > 
> 
> In AKS current minimum version is 1.22.6 so impossible to use k8s 1.20.x

You found any work around?

**Comment by 360DigiTMG-Training** [comment]
Hi Experts,

I am trying to set up Kubeflow on Azure using the official documentation of Kubeflow.
[https://www.kubeflow.org/docs/distributions/azure/

Encountered multiple errors:
az aks create -g <RESOURCE_GROUP_NAME> -n <NAME> -s <AGENT_SIZE> -c <AGENT_COUNT> -l <LOCATION> --generate-ssh-keys

Fails with error:
Values of identifierUris property must use a verified domain of the organization or its subdomain: 'https://90e9ea.Kubecluste-Kubeflowgroup-1760be.westus.cloudapp.azure.com'

I could not find any direct solution provided for the above error. I did found that registering the new domain in Azure Active Directory will resolve this issue. When i tried to do this, I found that my account has 'New Custom Domain' option disabled.

I did try to create the AKS cluster from Azure Portal (rather than CLI) and proceeded further with kubeflow setup.

At a later stage I encountered another error while configuring the kubeflow

command: kfctl apply -V -f ${CONFIG_URI}

error:

serviceaccount/application-controller-service-account unchanged

clusterrole.rbac.authorization.k8s.io/application-controller-cluster-role unchanged

clusterrolebinding.rbac.authorization.k8s.io/application-controller-cluster-role-binding unchanged

service/application-controller-service unchanged

statefulset.apps/application-controller-stateful-set configured

WARN[0210] Encountered error applying application application: (kubeflow.error): Code 500 with message: Apply.Run : [unable to recognize "/tmp/kout457914242": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1", unable to recognize "/tmp/kout457914242": no matches for kind "Application" in version "app.k8s.io/v1beta1"] filename="kustomize/kustomize.go:284"

WARN[0210] Will retry in 24 seconds. filename="kustomize/kustomize.go:285"

I could not find any solution for these issues. Request for any pointers to setup kubeflow on Azure.

Thankyou in advance

**Comment by Sharat-Innodatatics** [comment]
Hi Experts,

I am trying to set up Kubeflow on Azure using the official documentation of Kubeflow.
[https://www.kubeflow.org/docs/distributions/azure/

Encountered multiple errors:

az aks create -g <RESOURCE_GROUP_NAME> -n <NAME> -s <AGENT_SIZE> -c <AGENT_COUNT> -l <LOCATION> --generate-ssh-keys

Fails with error:
Values of identifierUris property must use a verified domain of the organization or its subdomain: 'https://90e9ea.Kubecluste-Kubeflowgroup-1760be.westus.cloudapp.azure.com'

I could not find any direct solution provided for the above error. I did find that registering the new domain in Azure Active Directory will resolve this issue. When I tried to do this, I found that my account has the 'New Custom Domain' option disabled.

I did try to create the AKS cluster from Azure Portal (rather than CLI) and proceeded further with the kubeflow setup.

At a later stage, I encountered another error while configuring the kubeflow

command: kfctl apply -V -f ${CONFIG_URI}

error:

serviceaccount/application-controller-service-account unchanged

clusterrole.rbac.authorization.k8s.io/application-controller-cluster-role unchanged

clusterrolebinding.rbac.authorization.k8s.io/application-controller-cluster-role-binding unchanged

service/application-controller-service unchanged

statefulset.apps/application-controller-stateful-set configured

WARN[0210] Encountered error applying application application: (kubeflow.error): Code 500 with message: Apply.Run : [unable to recognize "/tmp/kout457914242": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1", unable to recognize "/tmp/kout457914242": no matches for kind "Application" in version "app.k8s.io/v1beta1"] filename="kustomize/kustomize.go:284"

WARN[0210] Will retry in 24 seconds. filename="kustomize/kustomize.go:285"

I could not find any solution for these issues. Request for any pointers to setup kubeflow on Azure.

Thank you in advance
